eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -45,8 +45,51 @@ object LightningMessageCodecs {
   // (for something smarter see https://github.com/yzernik/bitcoin-scodec/blob/master/src/main/scala/io/github/yzernik/bitcoinscodec/structures/UInt64.scala)
   val uint64: Codec[Long] = int64.narrow(l => if (l >= 0) Attempt.Successful(l) else Attempt.failure(Err(s"overflow for value $l")), l => l)
 
+  val uint64L: Codec[Long] = int64L.narrow(l => if (l >= 0) Attempt.Successful(l) else Attempt.failure(Err(s"overflow for value $l")), l => l)
+
   val uint64ex: Codec[UInt64] = bytes(8).xmap(b => UInt64(b), a => a.toByteVector.padLeft(8))
 
+  // Bitcoin-style varint codec (CompactSize)
+  val varInt = Codec[Long](
```

Comment:
Same comment as above: should we use `UInt64`?

Have you considered naming it `compactsize`? apparently there seem to be confusion about multiple varint encodings in Bitcoin.

nit²: for some (probably bad) reason we use all lowercase codec names for codecs

Reply:
I'll think about it more but indeed it probably makes sense to use UInt64 here.

> Have you considered naming it compactsize?

It's true that it's missing a clear spec reference, I don't know yet what name will be used most...I tend to prefer varInt because CompactSize could be any type (doesn't mention Int) but I'll choose what comes out of the spec and will rename if we use compact size everywhere.

> nit²: for some (probably bad) reason we use all lowercase codec names for codecs

I was wondering why the all-lower case names :)
I'll do the same for consistency.